lib/thread.c

```diff
@@ -570,6 +570,7 @@ void thread_master_free(struct thread_master *m)
 	close(m->io_pipe[0]);
 	close(m->io_pipe[1]);
 	list_delete(m->cancel_req);
+	m->cancel_req = NULL;
```

Comment:
shouldn't be necessary, `XFREE` will set ptr to null

Reply:
It clearly does not.  Run a test if you want :)

Reply:
in this case `XFREE` is being called inside `list_delete() -> list_free()`. It will set the local `list` variable to NULL, `m->cancel_req` needs to be cleared manually.